c_emulator/riscv_sim.cpp

```diff
@@ -502,6 +379,46 @@ uint64_t load_sail(char *f, bool main_file)
   return entry;
 }
 
+uint64_t get_config_uint64(const char *key)
+{
+  std::stringstream keystream(key);
+  std::string part;
+  std::vector<const char *> parts;
+
+  while (std::getline(keystream, part, '.')) {
+    char *c_part = (char *)sail_malloc(part.size() + 1);
+    strncpy(c_part, part.c_str(), part.size() + 1);
+    parts.push_back(c_part);
+  }
+
+  sail_config_json json = sail_config_get(parts.size(), parts.data());
+
+  for (const char *c_part : parts) {
+    sail_free((void *)c_part);
+  }
```

Comment:
This function seems unnecessarily complex. How about the following and the callers can just do `get_config_uint64({"platform", "rom", "size"})`

```suggestion
uint64_t get_config_uint64(std::vector<const char *> keypath)
{
  sail_config_json json = sail_config_get(keypath.size(), keypath.data());
```

Reply:
Done